### DISCUSSION TITLE
Reset async loaded initial form values

### DISCUSSION THREAD

**Discussion by mattwcole** [mantinedev/mantine#3264] [Q&A] [answered]
Hello, the docs for the `useForm` hook provide an example of [setting the initial form values with an async request](https://mantine.dev/form/recipes/#set-initial-values-with-async-request). How can we make `form.reset()` restore the loaded initial values instead of the empty ones passed to the hook? Adding a call to `form.resetDirty()` after setting the initial values appears to have no effect.

```ts
useEffect(() => {
  loadInitialValues().then((values) => {
    form.setValues(values);
    form.resetDirty(); // Form is still rendered dirty
    form.resetTouched();
  });
}, []);
```

CodeSandbox [here](https://codesandbox.io/s/heuristic-bash-o10rcs?file=/src/MyForm.tsx).

**Comment by rtivital** [comment, marked as answer]
Specify new values in `form.resetDirty(values)` to update these values as well.

**Reply by mattwcole** [reply]
Thanks, this gets some of the way as `form.isDirty()` is now `false` as expected, but resetting the form still restores the original empty values, and not the loaded ones.

```ts
useEffect(() => {
  loadInitialValues().then((values) => {
    form.setValues(values);
    form.resetDirty(values); // Form is no longer dirty, but resetting it later does not restore these values
    form.resetTouched();
  });
}, []);
```

**Reply by rtivital** [reply]
Then you can call `form.setValues` with your values instead of `form.reset`. 
In this case, `form.reset` works correctly – it resets the form to initial values, not the values that you've set after the form was mounted.

**Reply by mattwcole** [reply]
I think I can make that work, thanks for the help, much appreciated!
